echopop/workflow/feat_hake.py

```diff
@@ -558,118 +561,306 @@
+# sex 
+# --------------------------------------------------------------------------------------------------
 
-biology.set_population_metrics(df_nasc=df_nasc_no_age1, 
-                               metrics=["abundance", "biomass", "biomass_density"],
-                               stratify_by="stratum_ks",
-                               df_average_weight=df_averaged_weight["all"])
+biology.compute_abundance(
+    dataset=df_nasc_no_age1,
+    stratify_by=["stratum_ks"],
+    group_by=["sex"],
+    exclude_filter={"sex": "unsexed"},
+    number_proportions=dict_df_number_proportion
+)
 
 # ==================================================================================================
-# Apportion age-1 vs age-2+ population estimates 
-# ----------------------------------------------
-# TODO: This apportionment step is required for kriging
+# Calculate (and apportion) biomass densities and biomass (from number density and abundance, 
+# respectively) for the overall transect dataset as well as for each sex
+# --------------------------------------------------------------------------------------------------
+
+biology.compute_biomass(
+    dataset=df_nasc_no_age1,
+    stratify_by=["stratum_ks"],
+    group_by=["sex"],
+    df_average_weight=df_averaged_weight,
+)
+
+# ==================================================================================================
+# Get proportions for each stratum specific to age-1
+# --------------------------------------------------
+
+# Age-1 NASC proportions
+age1_nasc_proportions = get_proportions.get_nasc_proportions_slice(
+    number_proportions=dict_df_number_proportion["aged"],
+    stratify_by=["stratum_ks"],
+    ts_length_regression_parameters={"slope": 20., 
+                                     "intercept": -68.},
+    include_filter = {"age_bin": [1]}
+)
+
+# Age-1 number proportions
+age1_number_proportions = get_proportions.get_number_proportions_slice(
+    number_proportions=dict_df_number_proportion["aged"],
+    stratify_by=["stratum_ks"],
+    include_filter = {"age_bin": [1]}
+)
+
+# Age-1 weight proportions
+age1_weight_proportions = get_proportions.get_weight_proportions_slice(
+    weight_proportions=dict_df_weight_proportion["aged"],
+    stratify_by=["stratum_ks"],
+    include_filter={"age_bin": [1]},
+    number_proportions=dict_df_number_proportion,
+    length_threshold_min=10.0,
+    weight_proportion_threshold=1e-10
+)
+
+# ==================================================================================================
+# Apply the calculated proportions to the abundance, biomass, and NASC estimates
+# ------------------------------------------------------------------------------
+
+df_nasc_no_age1_prt = apportion.partition_transect_data(
+    dataset=df_nasc_no_age1,
+    partition_dict={
+        "nasc": age1_nasc_proportions, 
+        "abundance": age1_number_proportions,
+        "biomass": age1_weight_proportions
+    },
+)
```

Comment:
I recommend the following changes to this function:
- rename to `remove_target_group`: the current function is unclear whether you are preserving or removing the target group
- make the "nasc", "abundance", "biomass" not optional -- i.e. the function will ALWAYS remove the target group (in this case age-1 fish) from columns with those names. This will avoid accidental omission that results in partially processed data

Reply:
Also, why is `df_nasc_no_age1_prt` not just `df_nasc_no_age1`? Without removing the age1 contribution in NASC, abundance, biomass, aren't the results in `df_nasc_no_age1` (after `invert_hake`, `compute_biomass`, and `compute_abundance`) wrong?